pallets/capacity/src/lib.rs

```diff
@@ -462,6 +480,49 @@ pub mod pallet {
 			Self::deposit_event(Event::EpochLengthUpdated { blocks: length });
 			Ok(())
 		}
+
+		/// Sets the target of the staking capacity to a new target.
+		/// This adds a chunk to `StakingAccountDetails.stake_change_unlocking chunks`, up to `T::MaxUnlockingChunks`.
+		/// The staked amount and Capacity generated by `amount` originally targeted to the `from` MSA Id is reassigned to the `to` MSA Id.
+		/// Does not affect unstaking process or additional stake amounts.
+		/// ### Errors
+		/// - [`Error::StakingAccountNotFound`] if origin does not have a staking account
+		/// - [`Error::MaxUnlockingChunksExceeded`] if `stake_change_unlocking_chunks` == `T::MaxUnlockingChunks`
+		/// - [`Error::StakerTargetRelationshipNotFound`] if `from` is not a target for Origin's staking account.
+		/// - [`Error::StakingAmountBelowMinimum`] if `amount` to retarget is below the minimum staking amount.
+		/// - [`Error::InsufficientStakingBalance`] if `amount` to retarget exceeds what the staker has targeted to `from` MSA Id.
+		/// - [`Error::InvalidTarget`] if `to` does not belong to a registered Provider.
+		#[pallet::call_index(4)]
+		#[pallet::weight(T::WeightInfo::unstake())]
+		pub fn change_staking_target(
+			origin: OriginFor<T>,
+			from: MessageSourceId,
+			to: MessageSourceId,
+			amount: BalanceOf<T>,
+		) -> DispatchResult {
```

Comment:
Do we want to ensure no dust is left in the current target? For example, if the amount staked from the "from" target goes below the minimum then the whole amount gets transferred or we emit an error.

Reply:
I'm actually not sure and I've waffled a bit on it.  For the StakingTargetDetails, probably.  For the CapacityDetails, no I don't think it should.

Reply:
Ok, I had to do some refactoring but I made it so that on `withdraw` a StakingTargetDetails will check to see if it's less than the minimum amount, and if so it sets the details to `default()` and returns the amounts that were in there.